lib/utils/userInputNumberFormatter.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/services.dart';
+
+class UserInputNumberFormatter extends TextInputFormatter{
+  @override
+  TextEditingValue formatEditUpdate(TextEditingValue oldValue, TextEditingValue newValue) {
```

Comment:
This is a classic case for a unit test - can you create a unit test that tests all possible things that could happen here?

E.g. validate

1,000.01
1,000
1.000.000,02 (spanish million-comma-zero-two)

And so on...  you could make the main algorithm a static method or something else that can easily be tested if that helps

Reply:
I would not have confidence in this code without unit test.

Reply:
Yes good Idea, I will create a unit test with all possible values/formats.  
I think this should cover most if not all cases:
95 pass
56.67 pass
45.56.45 pass
344..343.. pass 
34,5 pass 
46..474,38493.3627,384 pass
345,,,, pass 
453,343,343.676 pass 
456.454.345435 pass 
99,99,5545,45, pass

Reply:
this is good, needs to be tested in unit tests now